testing.go

```diff
@@ -41,29 +54,108 @@ func StartTest(t *testing.T) *Test {
 	})
 	span.SetBaggageItem("trace.kind", "test")
 
-	return &Test{
+	// Replaces stdout and stderr
+	stdOut := newStdIO(&os.Stdout)
+	stdErr := newStdIO(&os.Stderr)
+	log2.SetOutput(stdOut.writePipe)
+
+	test := &Test{
 		ctx:  ctx,
 		span: span,
 		t:    t,
+		stdOut: stdOut,
+		stdErr: stdErr,
+	}
+
+	// Starts stdIO pipe handlers
+	if test.stdOut != nil {
+		go stdIOHandler(test, test.stdOut, false)
 	}
+	if test.stdErr != nil {
+		go stdIOHandler(test, test.stdErr, true)
+	}
+
+	return test
 }
 
 func (test *Test) End() {
+	defer test.span.Finish()
+
 	if r := recover(); r != nil {
 		test.span.SetTag("test.status", "ERROR")
+		test.stdOut.restore(&os.Stdout)
+		test.stdErr.restore(&os.Stderr)
+		test.span.Finish()
 		_ = GlobalAgent.Flush()
 		panic(r)
 	}
-	test.span.Finish()
+
 	if test.t.Failed() {
 		test.span.SetTag("test.status", "FAIL")
 	} else if test.t.Skipped() {
 		test.span.SetTag("test.status", "SKIP")
 	} else {
 		test.span.SetTag("test.status", "PASS")
 	}
+
+	test.stdOut.restore(&os.Stdout)
+	test.stdErr.restore(&os.Stderr)
+	log2.SetOutput(os.Stderr)
```

Comment:
Is this the default value? Is there a way to save the original output target and restore that?

Reply:
for stdOut and stdError I store the original value and replacing with the pipe using the line: 
```stdOut := newStdIO(&os.Stdout)```

then in the restore, I just restore the original `*File`:
```test.stdOut.restore(&os.Stdout)``` 

that's the reason I pass the reference `&os.Stdout` instead just the value. A copy of the original `*File` is in the `stdOut` struct

For the `log.SetOutput` the default one is `os.Stderr` the problem is I didn't found a way to Get the current output in case the user already set a custom one.

Reply:
Forget it, I just saw a method in the `log.go` that gives me the current writer. `Writer()` method. I'll do the change.

Reply:
The writer method is not available in the static implementation.

And here is the default for the static one in `log.go`:
```go
var std = New(os.Stderr, "", LstdFlags)
```